Habitica/src/main/java/com/habitrpg/android/habitica/ui/activities/CreateChallengeActivity.java

```diff
@@ -0,0 +1,563 @@
+        bundle.putBoolean(TaskFormActivity.SET_IGNORE_FLAG, true);
+        bundle.putBoolean(TaskFormActivity.SHOW_TAG_SELECTION, false);
+        bundle.putBoolean(TaskFormActivity.SHOW_CHECKLIST, false);
+
+        if (HabiticaApplication.User != null && HabiticaApplication.User.getPreferences() != null) {
+            String allocationMode = HabiticaApplication.User.getPreferences().getAllocationMode();
+
+            bundle.putString(TaskFormActivity.USER_ID_KEY, HabiticaApplication.User.getId());
+            bundle.putString(TaskFormActivity.ALLOCATION_MODE_KEY, allocationMode);
+        }
+
+        Intent intent = new Intent(this, TaskFormActivity.class);
+        intent.putExtras(bundle);
+        intent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
+
+        startActivityForResult(intent, 1);
+    }
+
+    private void openNewTaskActivity(Task task) {
+        Bundle bundle = new Bundle();
+        bundle.putParcelable(TaskFormActivity.PARCELABLE_TASK, task);
+        bundle.putBoolean(TaskFormActivity.SAVE_TO_DB, false);
+        bundle.putBoolean(TaskFormActivity.SET_IGNORE_FLAG, true);
+        bundle.putBoolean(TaskFormActivity.SHOW_TAG_SELECTION, false);
+        bundle.putBoolean(TaskFormActivity.SHOW_CHECKLIST, false);
+
+        if (HabiticaApplication.User != null && HabiticaApplication.User.getPreferences() != null) {
+            String allocationMode = HabiticaApplication.User.getPreferences().getAllocationMode();
+
+            bundle.putString(TaskFormActivity.USER_ID_KEY, HabiticaApplication.User.getId());
+            bundle.putString(TaskFormActivity.ALLOCATION_MODE_KEY, allocationMode);
+        }
+
+        Intent intent = new Intent(this, TaskFormActivity.class);
+        intent.putExtras(bundle);
+        intent.setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
+
+        startActivityForResult(intent, 1);
+    }
+
+    private PostChallenge getChallengeData() {
+        PostChallenge c = new PostChallenge();
+
+        int locationPos = challenge_location_spinner.getSelectedItemPosition();
+        Group locationGroup = locationAdapter.getItem(locationPos);
+
+        if (challengeId != null) {
+            c.id = challengeId;
+        }
+
+        c.group = locationGroup.id;
+        c.name = create_challenge_title.getText().toString();
+        c.description = create_challenge_description.getText().toString();
+        c.shortName = create_challenge_tag.getText().toString();
+        c.prize = Integer.parseInt(create_challenge_prize.getText().toString());
+
+        return c;
+    }
+
+    private void createChallenge() {
+        PostChallenge c = getChallengeData();
+
+        List<Task> taskList = challengeTasks.getTaskList();
+        taskList.remove(addHabit);
+        taskList.remove(addDaily);
+        taskList.remove(addTodo);
+        taskList.remove(addReward);
+
+        challengeRepository.createChallenge(c, taskList).subscribe(createdChallenge -> {
```

Comment:
for the progress indicator you just mean `indeterminate`-state right? otherwise this would be even harder to add some callbacks between observables to show something like ` 1 of X ` progress

for the multiple presses, that should be simple, if its already running I'll prevent the next calls until its finished